bootloader/lkenv/lkenv.go

```diff
@@ -228,10 +230,13 @@ func (l *Env) Load() error {
 // LoadEnv loads the lk bootloader environment from the specified file. The
 // bootloader environment in the referenced file must be of the same version
 // that the Env object was created with using NewEnv.
+// The returned error may wrap os.ErrNotExist, so instead of using
+// os.IsNotExist, callers should use xerrors.Is(err,os.ErrNotExist) instead.
 func (l *Env) LoadEnv(path string) error {
 	f, err := os.Open(path)
 	if err != nil {
-		return fmt.Errorf("cannot open LK env file: %v", err)
+		fmtStr := "cannot open LK env file: %w"
+		return xerrors.Errorf(fmtStr, err)
```

Comment:
In 1.9 the `os.*Error` did not implement Unwrap(), so this won't work. OTOH, is it useful to wrap the error here instead of letting the caller use os.IsNotExist()?

Reply:
it is actually useful, it helped me debug / diagnose an error seeding that @kubiko had on Wednesday with something like:

```
error: cannot set recovery environment: cannot open LK env file: open /tmp/uc-db-builder-workdir/system-seed/boot/lk/snapbootsel.binbak: no such file or directory
```

the fact that there is only one place in the lkenv bootloader code we return an error of the form "cannot open LK env file" was useful to me here, if we returned a generic thing we would only see:

```
error: cannot set recovery environment: open /tmp/uc-db-builder-workdir/system-seed/boot/lk/snapbootsel.binbak: no such file or directory
```

in which case it's not clear which part of the lk bootloader functions we failed in

Reply:
It's quite unfortunate that go 1.9 is broken like this, I thought that's why we use `xerrors` though? anyways if it doesn't work on go 1.9 I'm inclined to just skip the specific test for this behavior on 1.9 due to the usefulness I explained above

Reply:
see other comment, but I added a workaround here, perhaps it is too invasive

Reply:
I'm no longer hard-coding which version of Go, we just inspect the error to see if it can do Unwrap, and if so then use it, if not return os.ErrNotExist directly